Userland/Libraries/LibGfx/Painter.cpp

```diff
@@ -623,9 +623,19 @@ void Painter::draw_rect_with_thickness(IntRect const& rect, Color color, int thi
     IntPoint p3 = { rect.location().x() + rect.width(), rect.location().y() + rect.height() };
     IntPoint p4 = { rect.location().x(), rect.location().y() + rect.height() };
 
-    draw_line(p1, p2, color, thickness);
+    if (thickness == 1) {
+        auto line_to_rect = [&](IntPoint p1, IntPoint p2) {
+            return IntRect::from_two_points(p1, p2.translated(1, 1));
+        };
+        fill_rect(line_to_rect(p1, p2.translated(-1, 0)), color);
+        fill_rect(line_to_rect(p2, p3), color);
+        fill_rect(line_to_rect(p4, p3.translated(-1, 0)), color);
+        fill_rect(line_to_rect(p4, p1), color);
```

Comment:
doesn't this overlap the rects from lines 1 and 3?

why fill_rect instead of draw_line for 1-thick lines anyways? special cases is where bugs hide etc

Reply:
I've redone this patch to fix the underlying issue in `draw_line()`. This fix was a little lazy :sweat_smile: